flake8_trio/visitors/visitors.py

```diff
@@ -0,0 +1,415 @@
+            if isinstance(n, ast.Name):
+                return n.id in startable_list
+            if isinstance(n, ast.Attribute):
+                return n.attr in startable_list
+            if isinstance(n, ast.Call):
+                return any(is_startable(nn, *startable_list) for nn in n.args)
+            return False
+
+        if (
+            self.aenter
+            and is_nursery_call(node.func, "start_soon")
+            and len(node.args) > 0
+            and is_startable(
+                node.args[0],
+                *STARTABLE_CALLS,
+                *self.options.startable_in_context_manager,
+            )
+        ):
+            self.error(node)
+
+
+# Checks that all async functions with a "task_status" parameter have a match in
+# --startable-in-context-manager. Will only match against the last part of the option
+# name, so may miss cases where functions are named the same in different modules/classes
+# and option names are specified including the module name.
+@error_class
+class Visitor114(Flake8TrioVisitor):
+    error_codes = {
+        "TRIO114": (
+            "Startable function {} not in --startable-in-context-manager parameter "
+            "list, please add it so TRIO113 can catch errors using it."
+        ),
+    }
+
+    def visit_AsyncFunctionDef(self, node: ast.AsyncFunctionDef):
+        if any(
+            isinstance(n, ast.arg) and n.arg == "task_status"
+            for n in self.walk(
+                *node.args.args, *node.args.posonlyargs, *node.args.kwonlyargs
+            )
+        ) and not any(
+            node.name == opt
+            for opt in (*self.options.startable_in_context_manager, *STARTABLE_CALLS)
+        ):
+            self.error(node, node.name)
+
+
+# Suggests replacing all `trio.sleep(0)` with the more suggestive
+# `trio.lowlevel.checkpoint()`
+@error_class
+class Visitor115(Flake8TrioVisitor):
+    error_codes = {
+        "TRIO115": "Use `trio.lowlevel.checkpoint()` instead of `trio.sleep(0)`.",
+    }
+
+    def visit_Call(self, node: ast.Call):
+        if (
+            get_matching_call(node, "sleep")
+            and len(node.args) == 1
+            and isinstance(node.args[0], ast.Constant)
+            and node.args[0].value == 0
+        ):
+            self.error(node)
+
+
+@error_class
+class Visitor116(Flake8TrioVisitor):
+    error_codes = {
+        "TRIO116": (
+            "trio.sleep() with >24 hour interval should usually be "
+            "`trio.sleep_forever()`"
+        ),
+    }
+
+    def visit_Call(self, node: ast.Call):
+        if get_matching_call(node, "sleep") and len(node.args) == 1:
+            arg = node.args[0]
+            if (
+                # `trio.sleep(math.inf)`
+                (
+                    isinstance(arg, ast.Attribute)
+                    and isinstance(arg.value, ast.Name)
+                    and arg.attr == "inf"
+                    and arg.value.id == "math"
```

Comment:
We might want to generalize this by dropping the `arg.value.id` check so that it catches e.g. `np.inf` too.